### PR TITLE
feat: add support for CI Network config in nocloud

### PR DESCRIPTION
**Pull request by ekarlso**
# Pull Request

## Acceptance

Please use the following checklist:

- [ ] you linked an issue (if applicable)
- [ ] you included tests (if applicable)
- [ ] you ran conformance (`make conformance`)
- [ ] you formatted your code (`make fmt`)
- [ ] you linted your code (`make lint`)
- [ ] you generated documentation (`make docs`)
- [ ] you ran unit-tests (`make unit-tests`)

> See `make help` for a description of the available targets.

### CODE REVIEW

**Review by smira — APPROVED** [review]
thank you, I made some small fixes

**Comment by TimJones** [comment]
Hi @ekarlso, thanks for your contribution. This seems to be `cloud-init` formatted network config, not `nocloud`.
See https://cloudinit.readthedocs.io/en/latest/reference/datasources/nocloud.html#id2 for `nocloud` metadata format, and you can see the first `network:` stanza is not part of the spec.

That said, many providers make the same mistake so there might be value in adding support for the `network:` part of the `cloud-init` spec as a separate platform.

File: internal/app/machined/pkg/runtime/v1alpha1/platform/nocloud/metadata.go [hunk 1]
```diff
@@ -43,6 +43,11 @@ const (
 	mnt                     = "/mnt"
 )
 
+// NetworkPlainConfig wraps nocloud network config to match cloud-init format.
+type NetworkPlainConfig struct {
```
Comment (TimJones) [inline internal/app/machined/pkg/runtime/v1alpha1/platform/nocloud/metadata.go:47]:
```suggestion
type NetworkCloudInitConfig struct {
```

**Comment by TimJones** [comment]
Actually I made some notes on changes that are minimal but make the difference more visible.

**Comment by smira** [comment]
@TimJones applied your suggestions

File: internal/app/machined/pkg/runtime/v1alpha1/platform/nocloud/testdata/metadata-v2-plain.yaml
Comment (TimJones) [inline internal/app/machined/pkg/runtime/v1alpha1/platform/nocloud/testdata/metadata-v2-plain.yaml:1]: Just a nit, but can this also be renamed `metadata-v2-cloudinit.yaml`

Comment (smira) [reply internal/app/machined/pkg/runtime/v1alpha1/platform/nocloud/testdata/metadata-v2-plain.yaml:1]: done!

**Pull request merged**